lib/src/cloud_firestore_mocks_base.dart

```diff
@@ -78,6 +79,24 @@ class MockFirestoreInstance extends Mock implements FirebaseFirestore {
     return await transactionHandler(transaction);
   }
 
+  dynamic deepFixDocumentReferenceProperty(dynamic propertyValue) {
+    if (propertyValue is MockDocumentReference) return propertyValue.toJson();
+
+    if (propertyValue is List) {
+      for (var i = 0; i < propertyValue.length; i++) {
+        propertyValue[i] = deepFixDocumentReferenceProperty(propertyValue[i]);
```

Comment:
Looks like `propertyValue` comes from `_docsData`, so this will accidentally overwrite the actual document value.